### PR TITLE
SIDM-9089 Dependency upgrade

### PR DESCRIPTION
**Pull request by nikola-naydenov-hmcts**

### CODE REVIEW

File: lib/applicationinsights.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+  "connectionString": "${APP_INSIGHTS_CONNECTION_STRING}",
+  "role": {
+    "name": "idam-health-checker"
+  },
+  "preview": {
+    "roleNameOverrides": [
+      {
+        "httpPathPrefix": "/health",
```
[the remark this answers is not in the document]
Comment (nikola-naydenov-hmcts) [reply lib/applicationinsights.json:9]: tbh I am not 100% sure what this setting does

[the remark this answers is not in the document]
Comment (jburke-idam) [reply lib/applicationinsights.json:9]: it gives a different "cloud_RoleName" value to these calls. Might not make sense in the health checker itself. In other apps you end up with "idam-api" and "idam-api (health)" so all the health calls don't get in the way of searching for "real" requests, but in this case there probably isn't that distinction

File: lib/applicationinsights.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      }
+    ],
+    "sampling": {
+      "overrides": [
+        {
+          "telemetryType": "request",
+          "attributes": [
+            {
+              "key": "http.url",
+              "value": "https?://[^/]+/health.*",
```
Comment (jburke-idam) [inline lib/applicationinsights.json:20]: this probably needs to be /admin/health too

File: src/main/java/uk/gov/hmcts/reform/idam/health/ApplicationConfig.java [hunk 3]
```diff
@@ -27,31 +16,4 @@ public TaskScheduler taskScheduler(){
         return threadPoolTaskScheduler;
     }
 
-    @Bean
-    public TelemetryInitializer telemetryInitializer() {
-        return new SpringBootTelemetryInitializer() {
-            @Override
-            public void initialize(Telemetry telemetry) {
-                super.initialize(telemetry);
-                telemetry.getContext().getCloud().setRoleInstance(appInstance);
```
[the remark this answers is not in the document]
Comment (nikola-naydenov-hmcts) [reply src/main/java/uk/gov/hmcts/reform/idam/health/ApplicationConfig.java:36]: image
Seems like the agent took care of that

File: lib/applicationinsights.json (hunk 2 above)
Comment (nikola-naydenov-hmcts) [reply lib/applicationinsights.json:20]: Right changed it

File: lib/applicationinsights.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (nikola-naydenov-hmcts) [reply lib/applicationinsights.json:9]: Sure changed it

**Review by kremi — APPROVED** [review]

**Pull request merged**